tools/apiview/parsers/rust-api-parser/ci.yml

```diff
@@ -0,0 +1,114 @@
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    stages:
+      - stage: 'Build'
+        variables:
+          - template: /eng/pipelines/templates/variables/image.yml
+          - name: NodeVersion
+            value: '18.x'
+          - name: ArtifactName
+            value: 'apiview'
+          - name: RustGeneratorDirectory
+            value: 'tools/apiview/parsers/rust-api-parser'
+        jobs:
+          - job: 'Build'
+
+            pool:
+              name: $(LINUXNEXTPOOL)
+              image: $(LINUXNEXTVMIMAGE)
+              os: linux
+
+            steps:
+              - task: NodeTool@0
+                inputs:
+                  versionSpec: '$(NodeVersion)'
+                displayName: 'Use NodeJS $(NodeVersion)'
+
+              - script: |
+                  npm install -g npm@8.16.0
+                displayName: "Install npm 8.16.0"
+
+              - script: |
+                  npm install
+                workingDirectory: $(RustGeneratorDirectory)
+                displayName: "Install npm packages Rust generator"
+
+              - script: |
+                  npm run-script build
+                workingDirectory: $(RustGeneratorDirectory)
+                displayName: "Build Rust generator"
+
+              - pwsh: |
+                  npm pack $(RustGeneratorDirectory)
+                  Copy-Item ./*.tgz $(Build.ArtifactStagingDirectory)
+                displayName: "Pack Rust generator"
+
+              - task: 1ES.PublishPipelineArtifact@1
+                inputs:
+                  targetPath: '$(Build.ArtifactStagingDirectory)'
+                  artifactName: $(ArtifactName)
+
+      - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
+        - stage: 'Release'
+          dependsOn: Build
+          condition: Succeeded()
+          variables:
+            - template: /eng/pipelines/templates/variables/image.yml
+            - name: ArtifactName
+              value: 'apiview'
+            - name: RustGeneratorDirectory
+              value: 'tools/apiview/parsers/rust-api-parser'
+          jobs:
+          - job: PublishPackage
+            displayName: 'Publish rust-genapi package to devops feed'
+            pool:
+              name: $(LINUXNEXTPOOL)
+              image: $(LINUXNEXTVMIMAGE)
+              os: linux
+            steps:
+            - download: current
+
+            - pwsh: |
+                $detectedPackageName=Get-ChildItem $(Pipeline.Workspace)/$(ArtifactName)/*.tgz
+                Write-Host "Detected package name: $detectedPackageName"
+                Write-Host "##vso[task.setvariable variable=PackageArtifactFullPath]$detectedPackageName"
+              displayName: "Find package artifact"
+
+            - pwsh: |
+                @"
+                registry=https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-rust/npm/registry/
```

Comment:
Should we publish to the `azure-sdk-for-js` registry since its a JS package

Reply:
```suggestion
                registry=https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-js/npm/registry/
```

Reply:
So the question is which ADO feed to publish the packages to?

I was looking at how the tools repo handles this (https://github.com/search?q=repo%3AAzure%2Fazure-sdk-tools%20https%3A%2F%2Fpkgs.dev.azure.com%2Fazure-sdk%2Fpublic%2F_packaging&type=code ) and it seems that today we either use the `azure-sdk-for-js` one or one called `azure-sdk-for-js-test-autorest` which feels honestly worse.

I don't have super strong opinions about it, and perhaps for now it will be easier to re-use the JS SDK one, but it would be nice to make something like "azure-tools-npm" in the future so we don't have the confusion about where the projects being published here actually live.

@weshaggard do you have any strong feelings about where we should be publishing this to?

Reply:
Updated to `azure-sdk-for-js` like Dozie suggested for now to not cause any disruptions.

If there are more thoughts, I'll update accordingly.

Reply:
In the general case we have been grouping packages by language so given this is a npm package I would suggest azure-sdk-for-js dev feed as well.

Reply:
thanks Wes